mtags/src/main/scala-3/scala/meta/internal/pc/CompletionProvider.scala

```diff
@@ -0,0 +1,169 @@
+      // then `_root_` would appear as a completion result in the code `foobar(_<COMPLETE>)`
+      defn.RootPackage,
+      // NOTE(gabro) valueOf was added as a Predef member in 2.13. We filter it out since is a niche
+      // use case and it would appear upon typing 'val'
+      defn.ValueOfClass.info.member(nme.valueOf).symbol
+    ).flatMap(_.alternatives.map(_.symbol)).toSet
+
+    private def computeRelevancePenalty(
+      sym: Symbol
+    ): Int = {
+        import MemberOrdering._
+        var relevance = 0
+        // local symbols are more relevant
+        if (!sym.isLocalToBlock) relevance |= IsNotLocalByBlock
+        // symbols defined in this file are more relevant
+        if (sym.span != Spans.NoSpan || sym.is(Package))
+        relevance |= IsNotDefinedInFile
+        // fields are more relevant than non fields
+        if (sym.isField) relevance |= IsNotGetter
+        // non-inherited members are more relevant
+        //if (isInherited) relevance |= IsInherited
+        // symbols whose owner is a base class are less relevant
+        if (sym.owner == defn.AnyClass || sym.owner == defn.ObjectClass)
+        relevance |= IsInheritedBaseMethod
+        // symbols not provided via an implicit are more relevant
+        if (sym.is(Implicit)) relevance |= IsImplicitConversion
+        if (sym.is(Package)) relevance |= IsPackage
+        // accessors of case class members are more relevant
+        if (!sym.is(CaseAccessor)) relevance |= IsNotCaseAccessor
+        // public symbols are more relevant
+        if (!sym.isPublic) relevance |= IsNotCaseAccessor
+        // synthetic symbols are less relevant (e.g. `copy` on case classes)
+        if (sym.is(Synthetic)) relevance |= IsSynthetic
+        if (sym.isDeprecated) relevance |= IsDeprecated
+        if (isEvilMethod(sym.name)) relevance |= IsEvilMethod
+        relevance
+    }
+
+    private lazy val isEvilMethod: Set[Name] = Set[Name](
+      nme.notifyAll_,
+      nme.notify_,
+      nme.wait_,
+      nme.clone_,
+      nme.finalize_
+    )
+
+    private def completionOrdering(
+      ctx: Context,
+      position: SourcePosition
+  ): Ordering[Completion]
+    = new Ordering[Completion] {
+      implicit val context: Context = ctx
+      def compareLocalSymbols(o1: Completion, o2: Completion): Int = {
+        val s1 = o1.sym
+        val s2 = o2.sym
+        if (s1.isLocal && s2.isLocal) {
+          if (s1.line > s2.line) -1
+          else 1
+        } else {
+          0
+        }
+      }
+      override def compare(o1: Completion, o2: Completion): Int = {
+        val byLocalSymbol = compareLocalSymbols(o1, o2)
+          if (byLocalSymbol != 0) byLocalSymbol
+          else {
+            val byRelevance = Integer.compare(
+              computeRelevancePenalty(o1.sym),
+              computeRelevancePenalty(o2.sym)
+            )
+            if (byRelevance != 0) byRelevance
+            else {
```

Comment:
it would be awesome if we have `byFuzzy` comparison between `byRelevance` and `byIdentifier` as scala2 does
https://github.com/scalameta/metals/blob/4da9f0e1c68380bd9b47b01609ee30b0446995b0/mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala#L173-L182

Reply:
Yes, I skipped that one because I previously failed to obtain query for completion, but now it's back